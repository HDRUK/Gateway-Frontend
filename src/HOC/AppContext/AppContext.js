import React, { Component } from "react";
import PropTypes from "prop-types";
import hdruk_logo_black from "../../assets/hdruk_black.png";

export const AppContext = React.createContext();

export const AppContextConsumer = AppContext.Consumer;

class AppContextProvider extends Component {
    newsItems = {
        newsItemOne: {
            image: "",
            description: "HDR release news of new partners in exciting new digital project",
            readMore: "http://localhost:3000"
        },
        newsItemTwo: {
            image: "",
            description: "HDR release news of new partners in exciting new digital project",
            readMore: "http://localhost:3000"
        },
        newsItemThree: {
            image: "",
            description: "HDR release news of new partners in exciting new digital project",
            readMore: "http://localhost:3000"
        }
    };

    images = {
        logoHDR: hdruk_logo_black
    };

    filter = [];
    textItems = { searchHeader: "What health data do you need?" };

    searchPageStates = {
        form: "form",
        results: "results"
    };

    itemRef = React.createRef();

    state = {
        counter: 0,
        searchPageState: this.searchPageStates.form,
        modalVisibility: false,
        filterLocation: 0,
        filterId: 0
    };

    returnSearchResults = event => {
        if (event.key === "Enter") {
            this.setState({
                searchPageState: this.searchPageStates["results"]
            });
        }
    };

    outsideRange = (number, target, range) => {
        return number < target - range || number > target + range;
    };

    setFilterLocation = () => {
        this.itemRef.current &&
            this.outsideRange(this.itemRef.current.getBoundingClientRect().y, this.state.filterLocation, 11) &&
            this.setState({ filterLocation: this.itemRef.current.getBoundingClientRect().y });
    };

    setFilterId = props => {
        this.setState({ filterId: props });
    };

    counterFunc = () => {
        this.setState({
            counter: this.state.counter + 1
        });
    };

    addFilter = props => {
        this.filters.append(props);
    };

    toggleModal = () => {
        this.setState(
            {
                modalVisibility: !this.state.modalVisibility
            },
            () =>
                this.state.modalVisibility
                    ? document
                          .getElementById("main-side-nav")
                          .childNodes[1].addEventListener("scroll", this.setFilterLocation)
                    : document
                          .getElementById("main-side-nav")
                          .childNodes[1].removeEventListener("scroll", this.setFilterLocation)
        );
    };

    render() {
        return (
            <AppContext.Provider
                value={{
                    state: this.state,
                    counterFunc: this.counterFunc,
                    newsItems: this.newsItems,
                    images: this.images,
                    textItems: this.textItems,
                    returnSearchResults: this.returnSearchResults,
                    setFilterLocation: this.setFilterLocation,
                    toggleModal: this.toggleModal,
                    setFilterId: this.setFilterId,
                    itemRef: this.itemRef
                }}
            >
                {this.props.children}
            </AppContext.Provider>
        );
    }
}

AppContextProvider.propTypes = {
    children: PropTypes.node
};

export default AppContextProvider;
