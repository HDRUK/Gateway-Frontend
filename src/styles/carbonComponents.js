import {
    Button,
    UnorderedList,
    Header,
    SideNavLinkText,
    Search,
    SideNav,
    Dropdown,
    Accordion,
    AccordionItem,
    SideNavItems,
    Checkbox,
    DatePicker,
    DatePickerInput,
    Loading,
    InlineLoading,
    Modal,
    Form,
    TextInput,
    FormGroup,
    FormLabel,
    TextArea,
    Tag,
    Tabs,
    Tab
} from "carbon-components-react";
import styled, { css } from "styled-components";

// HDR Imported carbon components ..
export const HDRTabs = Tabs;
export const HDRTab = Tab;

// Styled carbon components
export const StyledButton = styled(Button)`
    padding: 15px;
    width: 215px;
    height: 54px;
`;
export const FilterButton = styled(Button)`
    width: calc(100% + 4.6rem);
    margin-left: -1rem;
    min-height: 1rem;
    height: 2rem;
    padding: 0px 0px 0px 0.635rem;
    bottom: -1.75rem;
    margin-top: -1rem;
`;

export const ParagraphBullets = styled(UnorderedList)`
    padding-top: 5px;
`;

export const AccordionBlock = styled(Accordion)`
    width: 90%;
    align-self: center;
    margin: 1rem 0;
`;

export const AccordionElement = styled(AccordionItem)`
    position: relative;
    ${props =>
        props.modal === "true" &&
        props.open &&
        css`
            > div {
                padding: 0 !important;
            }
            * svg {
                transform: rotate(0deg) !important;
            }
        `}
`;

export const HeaderWrapper = styled(Header)`
    height: 4rem;
    max-width: 100%;
    position: unset;
    background: linear-gradient(135deg, ${p => p.theme.colors.primary}, ${p => p.theme.colors.secondary});
    border-bottom: 0;
    color: white;
`;

export const SideNavText = styled(SideNavLinkText)`
    font-size: 1.2rem !important;
`;

export const NavItems = styled(SideNavItems)`
    flex: inherit;
    overflow: visible !important;
`;

export const CheckboxItem = styled(Checkbox)`
    margin: 5px 10px 0px 0px;
`;

export const SearchBar = styled(Search)``;

export const DropdownFilter = styled(Dropdown)`
    min-width: 15.625rem;
`;

export const DateSelector = styled(DatePicker)`
    flex-direction: column;
`;

export const DateInput = styled(DatePickerInput)``;

export const MainSideNav = styled(SideNav)`
    display: inline-flex;
    position: absolute;
    background-color: ${p => p.theme.colors.white};
    border-right: 0.0625rem solid ${p => p.theme.colors.border};
    font-size: 1.2rem;
    overflow: auto;
    overflow-x: hidden;
    z-index: 1;
`;

export const CenterLoading = styled(Loading)`
    margin: 0 auto 0;
`;

export const RightSmallInlineLoading = styled(InlineLoading)`
    width: unset;
    float: right;
    font-size: 0.5rem;
`;

export const SaveSearchButton = styled(Button)`
    width: 90%;
    margin: 0.5rem auto 0;
    padding: 0 1rem 0 1rem;

    :disabled {
        background-color: ${p => p.theme.colors.darkBlue};
        border-color: ${p => p.theme.colors.darkBlue};
        color: ${p => p.theme.colors.white};

        :hover {
            background-color: ${p => p.theme.colors.darkBlue};
            border-color: ${p => p.theme.colors.darkBlue};
            color: ${p => p.theme.colors.white};
            outline-color: ${p => p.theme.colors.darkBlue};
        }
    }
`;

export const CustomTag = styled(Tag)`
    height: unset;
    min-height: 1.5rem;
`;

//NEW STYLES
export const NewStyledButton = styled(Button)`
    height: 2rem;
    padding: calc(0.875rem - 3px) 30px calc(0.875rem - 3px) 30px;
    background-color: ${p => p.theme.colors.primaryButton};
    :hover {
        background-color: ${p => p.theme.colors.primaryHover};
    }
`;

export const StyledForm = styled(Form)``;
export const StyledTextInput = styled(TextInput)`
    border: none;
`;
export const StyledTextArea = styled(TextArea)`
    border: none;
    :disabled {
        color: #c6c6c6;
    }
`;

export const StyledModal = styled(Modal)`
    .bx--modal-container {
        @media (min-width: 60rem) {
            width: 35%;
            max-width: 768px;
            max-height: 90%;
            height: auto;
            border-radius: 0.2rem;
        }
    }
    .bx--btn--primary {
        background-color: #2fbb93;
        right: 0;
        :hover {
            background-color: #249172;
        }
    }
    .bx--btn--secondary {
        background-color: #f4f4f4;
        border-color: #2fbb93;
        border-width: 1px;
        color: #2fbb93;
        :hover {
            border-color: #249172;
            color: #249172;
        }
    }
    .bx--modal-footer {
        background-color: transparent;
    }
    .bx--modal-footer button.bx--btn {
        height: 0.6rem;
        padding: 0rem 1rem 0rem 1rem;
        font-weight: 700;
        position: absolute;
        flex: 0;
        margin-left: 2rem;
        margin-right: 2rem;
    }
`;

export const StyledFormGroup = styled(FormGroup)``;

export const StyledFormLabel = styled(FormLabel)``;
