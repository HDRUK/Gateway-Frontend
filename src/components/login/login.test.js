import React from "react";
import { create } from "react-test-renderer";
import Login from "./Login";
import { MemoryRouter } from "react-router-dom";
import {
    CenterBlock,
    DarkText,
    Heading,
    MediumSpace,
    SmallSpace,
    LargeSpace,
    LinkNoDecoration
} from "../../styles/styles.js";
import { StyledButton } from "../../styles/carbonComponents.js";

import { AppContext } from "../../HOC/AppContext/AppContext";
import context from "../../__mocks__/AppContextMock.js";

const loginText = {
    headingText: "Log in to access our datasets",
    loginButton: "Log out",
    continueButton: "Continue without logging in"
};

describe("<Login> ", () => {
    let renderedComponent;
    let renderedOutput;

    beforeEach(() => {
        renderedComponent = create(
            <MemoryRouter>
                <AppContext.Provider value={context}>
                    <Login />
                </AppContext.Provider>
            </MemoryRouter>
        );
        renderedOutput = renderedComponent.root;
    });

    describe("when the login block is rendered", () => {
        it("should have the correct destination for the continue button", () => {
            const link = renderedOutput.findAllByType(LinkNoDecoration);
            const continueDestination = link[0].props.to;
            expect(continueDestination).toBe("/search");
        });
        it("should render the correct components", () => {
            const centerBlocks = renderedOutput.findAllByType(CenterBlock);
            const darkText = centerBlocks[0].findByType(DarkText);
            const darkTextChildren = darkText.props.children;
            expect(darkTextChildren).toHaveLength(4);
            expect(darkTextChildren[0].type).toEqual(Heading);
            expect(darkTextChildren[1].type).toEqual(MediumSpace);
            expect(darkTextChildren[2].type).toEqual(CenterBlock);
            expect(darkTextChildren[3].type).toEqual(LargeSpace);

            expect(darkTextChildren[0].props.children).toEqual(loginText.headingText);

            const centerBlock2Content = darkTextChildren[2].props.children;
            expect(centerBlock2Content).toHaveLength(3);
            expect(centerBlock2Content[0].type).toBe("a");
            const primaryButton = centerBlock2Content[0].props.children;
            expect(primaryButton.props.kind).toEqual("primary");
            expect(primaryButton.props.children).toEqual(loginText.loginButton);

            expect(centerBlock2Content[1].type).toEqual(SmallSpace);
            expect(centerBlock2Content[2].type).toEqual(LinkNoDecoration);
            expect(centerBlock2Content[2].props.to).toEqual("/search");

            const secondaryButton = centerBlock2Content[2].props.children;
            expect(secondaryButton.type).toEqual(StyledButton);
            expect(secondaryButton.props.kind).toEqual("secondary");
            expect(secondaryButton.props.children).toEqual(loginText.continueButton);
        });
    });
});
