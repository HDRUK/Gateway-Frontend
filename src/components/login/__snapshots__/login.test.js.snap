// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login>  should test the Login Snapshot 1`] = `
<div
  className="sc-htoDjs dtAbCu"
>
  <text
    className="sc-EHOje cUVptH"
  >
    <h1
      className="sc-bdVaJa hSReXH"
    >
      Log in to access our datasets
    </h1>
    <div
      className="sc-iwsKbI kOWQtT"
    />
    <div
      className="sc-htoDjs dtAbCu"
    >
      <button
        className="sc-jKJlTe gcTxff bx--btn bx--btn--primary"
        disabled={false}
        tabIndex={0}
        type="button"
      >
        Log in
      </button>
      <div
        className="sc-dnqmqq dlGTFF"
      />
      <a
        href="/search"
        onClick={[Function]}
      >
        <button
          className="sc-jKJlTe gcTxff bx--btn bx--btn--secondary"
          disabled={false}
          tabIndex={0}
          type="button"
        >
          Continue without logging in
        </button>
      </a>
    </div>
    <div
      className="sc-gZMcBi jYAjms"
    />
  </text>
</div>
`;
